code/game/objects/items/weapons/storage/bags.dm

```diff
@@ -460,6 +460,20 @@
 	return ..()
 
 
+/obj/item/storage/bag/tray/cookies_tray/New() /// By Azule Utama, thank you a lot!
+	..()
+	for(var/i in 1 to 6)
+		var/obj/item/C = new /obj/item/reagent_containers/food/snacks/cookie(src)
+		handle_item_insertion(C)    // Done this way so the tray actually has the cookies visible when spawned
+	rebuild_overlays()
+
+/obj/item/storage/bag/tray/sugarcookies_tray/New() /// By Azule Utama, thank you a lot!

```

Comment:
I'd suggest making this a subtype of the cookie tray, then adding a var to the tray that tells you what kind of cookie is meant to be created on the tray, then just spawning that var in the for loop. It'll save you copy-pasting code, and is also more OOP.

Reply:
Sounds like a good idea, I'm kinda new to all this coding thing, so I just read what's already done and try to make something out of it. So I can try.

Reply:
Aye, no worries. It's all part of the learning experience - a good rule of thumb is that if you end up with the exact same code in two different places you can probably make a proc that applies to both cases. Good luck!

Reply:
I did it, so it's fixed. I'm doing some final tests to all the suggestions you made and I'll change the files again.